server/market/integ/booker_matcher_test.go

```diff
@@ -1018,34 +1054,49 @@ func TestMatchWithBook_everything_multipleQueued(t *testing.T) {
 	epochQueueInit := make([]order.Order, len(epochQueue))
 	copy(epochQueueInit, epochQueue)
 
+	// var shuf []int
+	// matcher.ShuffleQueue(epochQueue)
+	// for i := range epochQueue {
+	// 	for j := range epochQueueInit {
+	// 		if epochQueue[i].ID() == epochQueueInit[j].ID() {
+	// 			shuf = append(shuf, j)
+	// 			t.Logf("%d: %p", j, epochQueueInit[j])
+	// 			continue
+	// 		}
+	// 	}
+	// }
+	// t.Logf("%#v", shuf)
+
 	// Apply the shuffling to determine matching order that will be used.
 	// matcher.ShuffleQueue(epochQueue)
 	// for i := range epochQueue {
 	// 	t.Logf("%d: %p, %p", i, epochQueueInit[i], epochQueue[i])
 	// }
-	// Shuffles to  [16, 6, 3, 14, 7, 12, 9, 13, 11, 1, 4, 10, 5, 2, 17, 8, 15, 0]
-	// 16 is a cancellation for a an epoch order not yet inserted (failed)
-	// 6  limit sell 4 lots matches 1 @ 450, 3 @ 430 (passed)
-	// 3  limit buy misses (failed)
-	// 14 cancel misses because that order matched epoch order 6 already (failed)
-	// 7  limit sell unfilled insert (insert, partial)
-	// 12 cancel order matches (passed)
-	// 9  market sell 4 lots matches 3 @ 430, 1 @ 400 (passed)
-	// 13 cancel order misses because it's order matched already (failed)
-	// 11 market buy 2 lots matches 1 @ 455, 1 @ 460 (passed)
-	// 1  limit buy 1 insert unfilled (partial, inserted)
-	// 4  limit sell 1 matches epoch order 1 just inserted (passed)
-	// 10 market buy 1 lot fills 1 @ 460 (passed)
+	// Shuffles to [9, 5, 6, 2, 7, 17, 14, 10, 0, 3, 8, 13, 12, 15, 11, 4, 1, 16]
```

Comment:
Unfortunately, brute-forcing a time offset as a nonce wasn't feasible for a queue this size, so I had to redo it.